app/src/main/java/swati4star/createpdf/util/PdfToImages.java

```diff
@@ -48,13 +74,18 @@ protected Void doInBackground(Void... voids) {
         // Render pdf pages as bitmap
         ParcelFileDescriptor fileDescriptor = null;
         try {
-            if (mUri != null)
-                // resolve pdf file path based on uri
-                fileDescriptor = ((PdfToImageFragment) mExtractImagesListener).getContext()
-                        .getContentResolver().openFileDescriptor(mUri, "r");
-            else if (mPath != null)
-                // resolve pdf file path based on relative path
-                fileDescriptor = ParcelFileDescriptor.open(new File(mPath), MODE_READ_ONLY);
+            if (mDecryptedPath != null)
+                fileDescriptor = ParcelFileDescriptor.open(new File(mDecryptedPath), MODE_READ_ONLY);
```

Comment:
And @vyankatesh24 The only problem i am facing is that the `mDecryptedPath` is coming `null` the first time. I mean how sync this after `mDecryptedPath = mPdfEncryptionUtility.removePasswordForImages(path, null);` gets executed? The file is indeed getting decrypted and a new decrypted path with a new path is getting created but as the processes are not in sync, It seems `doInBackground` works ahead. But It should execute after `onPreExecute` right? @vyankatesh24 Please check through and let me know.